### PR TITLE
First Part of the Class Implementation

### PR DESCRIPTION
**Pull request by askabderon**
implementation of the first part of the issue: https://github.com/bounswe/bounswe2022group7/issues/382
It runs without error and I checked the database to make sure about whether associations are implemented correctly or not.

### CODE REVIEW

**Review by sabrimete — APPROVED** [review]

File: backend/src/main/kotlin/com/group7/artshare/entity/RegisteredUser.kt [hunk 1]
```diff
@@ -38,16 +38,16 @@ class RegisteredUser(
     @Column
     var xp: Double = 0.0
 
-    @ManyToMany(mappedBy = "followers", cascade = [CascadeType.ALL])
+    @ManyToMany(mappedBy = "followers", cascade =  [CascadeType.MERGE, CascadeType.PERSIST])
     var following: Set<RegisteredUser> = HashSet()
```
Comment (sabrimete) [inline backend/src/main/kotlin/com/group7/artshare/entity/RegisteredUser.kt:42]: I think we should use MutableSet

File: backend/src/main/kotlin/com/group7/artshare/entity/RegisteredUser.kt [hunk 2]
```diff
@@ -94,13 +94,27 @@ class RegisteredUser(
...
+    var unreadNotifications: MutableSet<Notification> = mutableSetOf()
 
     //TODO discussion post
     //TODO past reply past posts
-    //TODO read notifications
-    //TODO unread notifications
-    //TODO current bids
 
+    @OneToMany(orphanRemoval = true, cascade = [CascadeType.ALL])
+    var currentBids: List<Bid> = ArrayList()
```
Comment (sabrimete) [inline backend/src/main/kotlin/com/group7/artshare/entity/RegisteredUser.kt:117]: MutableList may be required.

**Review by sabrimete — CHANGES_REQUESTED** [review]
I think changing datatypes with their mutable ones is required

File: backend/src/main/kotlin/com/group7/artshare/entity/RegisteredUser.kt (hunk 2 above)
Comment (askabderon) [reply backend/src/main/kotlin/com/group7/artshare/entity/RegisteredUser.kt]: You are right, updated with the latest commit.

**Comment by askabderon** [comment]
MutableSet is tested in the DataProp file. I also tested nested possible nested json back and added the necessary indications to prevent that. It seems there is no error on the endpoints:

> [
    {
        "id": 7,
        "artItemInfo": {
            "id": 8,
            "name": "Starry Night",
            "description": "A reflection of my soul from a fuzzy night.",
            "category": "Classical",
            "imageUrl": "https://media.overstockart.com/optimized/cache/data/product_images/VG485-1000x1000.jpg",
            "labels": "[\"classical\", \"painting\"]"
        },
        "creator": {
            "id": 9,
            "name": "Bridget",
            "surname": "Jones"
        },
        "creationDate": "2022-11-07T22:04:39.000+00:00",
        "owner": {
            "accountInfo": {
                "email": "email@email.com",
                "username": "janedoe",
                "password": "31415926",
                "id": 2,
                "name": "Jane",
                "surname": "Doe",
                "country": "England",
                "dateOfBirth": null,
                "profilePictureUrl": "https://images.pexels.com/photos/733872/pexels-photo-733872.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500"
            },
            "authorities": [],
            "userId": 1,
            "isVerified": false,
            "level": 3,
            "xp": 0.0,
            "following": [],
            "followers": [],
            "blockedUsers": [],
            "blockedBy": [],
            "isBanned": false,
            "allPhysicalExhibitions": [],
            "allOnlineGalleries": [],
            "bookmarkedArtItems": [
                {
                    "id": 10,
                    "artItemInfo": {
                        "id": 11,
                        "name": "Mediterranean Landscape",
                        "description": "Emotional hurricane on a dull landscape, too dull for an artistic soul as mine.",
                        "category": "Classical",
                        "imageUrl": "https://uploads4.wikiart.org/images/pablo-picasso/mediterranean-landscape-1952.jpg!Large.jpg",
                        "labels": "[\"classical\", \"painting\"]"
                    },
                    "creator": {
                        "id": 13,
                        "name": "Michael",
                        "surname": "Phelps"
                    },
                    "creationDate": "2022-11-07T22:04:39.000+00:00",
                    "owner": {
                        "accountInfo": {
                            "email": "email3@email.com",
                            "username": "janettdoe",
                            "password": "31415976",
                            "id": 6,
                            "name": "Janett",
                            "surname": "Doe",
                            "country": "Katalan",
                            "dateOfBirth": null,
                            "profilePictureUrl": "https://images.pexels.com/photos/733872/pexels-photo-733872.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500"
                        },
                        "authorities": [],
                        "userId": 5,
                        "isVerified": false,
                        "level": 3,
                        "xp": 0.0,
                        "following": [],
                        "followers": [],
                        "blockedUsers": [],
                        "blockedBy": [],
                        "isBanned": false,
                        "allPhysicalExhibitions": [],
                        "allOnlineGalleries": [],
                        "bookmarkedArtItems": [],
                        "bookmarkedPhysicalExhibitions": [],
                        "bookmarkedOnlineGalleries": [],
                        "readNotifications": [],
                        "unreadNotifications": [],
                        "currentBids": [],
                        "commentList": [],
                        "password": "31415976",
                        "enabled": true,
                        "username": "janettdoe",
                        "accountNonExpired": true,
                        "accountNonLocked": true,
                        "credentialsNonExpired": true,
                        "email": "email3@email.com"
                    },
                    "onAuction": false,
                    "auction": {
                        "id": 12,
                        "auctionInfo": null,
                        "creator": null,
                        "isActive": true,
                        "markedAsSold": false,
                        "bids": []
                    },
                    "lastPrice": 0.0,
                    "commentList": [
                        {
                            "itemId": 14,
                            "body": "Shclecht!",
                            "createdAt": "2022-11-07T22:04:39.000+00:00"
                        }
                    ]
                }
            ],
            "bookmarkedPhysicalExhibitions": [],
            "bookmarkedOnlineGalleries": [],
            "readNotifications": [],
            "unreadNotifications": [],
            "currentBids": [],
            "commentList": [],
            "password": "31415926",
            "enabled": true,
            "username": "janedoe",
            "accountNonExpired": true,
            "accountNonLocked": true,
            "credentialsNonExpired": true,
            "email": "email@email.com"
        },
        "onAuction": false,
        "auction": null,
        "lastPrice": 0.0,
        "commentList": [
            {
                "itemId": 15,
                "body": "Wonderful! I feel goosebumps!",
                "createdAt": "2022-11-07T22:04:39.000+00:00"
            },
            {
                "itemId": 16,
                "body": "Amazing!",
                "createdAt": "2022-11-07T22:04:39.000+00:00"
            }
        ]
    },
    {
        "id": 10,
        "artItemInfo": {
            "id": 11,
            "name": "Mediterranean Landscape",
            "description": "Emotional hurricane on a dull landscape, too dull for an artistic soul as mine.",
            "category": "Classical",
            "imageUrl": "https://uploads4.wikiart.org/images/pablo-picasso/mediterranean-landscape-1952.jpg!Large.jpg",
            "labels": "[\"classical\", \"painting\"]"
        },
        "creator": {
            "id": 13,
            "name": "Michael",
            "surname": "Phelps"
        },
        "creationDate": "2022-11-07T22:04:39.000+00:00",
        "owner": {
            "accountInfo": {
                "email": "email3@email.com",
                "username": "janettdoe",
                "password": "31415976",
                "id": 6,
                "name": "Janett",
                "surname": "Doe",
                "country": "Katalan",
                "dateOfBirth": null,
                "profilePictureUrl": "https://images.pexels.com/photos/733872/pexels-photo-733872.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500"
            },
            "authorities": [],
            "userId": 5,
            "isVerified": false,
            "level": 3,
            "xp": 0.0,
            "following": [],
            "followers": [],
            "blockedUsers": [],
            "blockedBy": [],
            "isBanned": false,
            "allPhysicalExhibitions": [],
            "allOnlineGalleries": [],
            "bookmarkedArtItems": [],
            "bookmarkedPhysicalExhibitions": [],
            "bookmarkedOnlineGalleries": [],
            "readNotifications": [],
            "unreadNotifications": [],
            "currentBids": [],
            "commentList": [],
            "password": "31415976",
            "enabled": true,
            "username": "janettdoe",
            "accountNonExpired": true,
            "accountNonLocked": true,
            "credentialsNonExpired": true,
            "email": "email3@email.com"
        },
        "onAuction": false,
        "auction": {
            "id": 12,
            "auctionInfo": null,
            "creator": null,
            "isActive": true,
            "markedAsSold": false,
            "bids": []
        },
        "lastPrice": 0.0,
        "commentList": [
            {
                "itemId": 14,
                "body": "Shclecht!",
                "createdAt": "2022-11-07T22:04:39.000+00:00"
            }
        ]
    }
]

**Review by sabrimete — APPROVED** [review]
LGTM

**Pull request merged**